src/pages/MatchEdition/EditMatch.tsx

```diff
@@ -0,0 +1,135 @@
+import { motion } from 'framer-motion'
+import dayjs from 'dayjs'
+import { useForm } from 'react-hook-form'
+import { Box, Stack } from '@mui/material'
+import { useParams } from 'react-router-dom'
+import CircularProgress from '@mui/material/CircularProgress'
+import { useEffect, useState } from 'react'
+import pick from 'lodash.pick'
+import { Match as MatchType } from '../../types/Match'
+import { useMatch, useMatchUpdate } from '../../queries/match'
+
+import { MatchParticipants } from '../../components/Match/MatchParticipants'
+import { MatchDetails } from '../../components/Match/MatchDetails'
+
+import { animationProps } from '../../styles/animation'
+import { FabSubmit } from '../../components/FabSubmit'
+import { getErrorMessage } from '../../utils/api'
+import { Alert } from '../../components/Alert'
+
+export const MatchEdition = () => {
+  const { id } = useParams()
```

Comment:
You can pass a generic type parameter to `useParams` to define what is the type of the params you are expecting.

Crazy idea for the future: maybe include a modified version of the `useParams` hook inside your `RouteSpec`, so that you can `import matchEdition from '../routeSpecs'` and use `matchEdition.useParams`, and the return type will already be typed